igc.h

```diff
@@ -161,4 +262,124 @@ class IgcFormat : public Format
+  */
+
+template <typename T,
+          typename = typename std::enable_if<std::is_floating_point<T>::value ||
+                                             std::is_same<T, int>::value>::type>
+struct igc_fsdata : public FormatSpecificData {
+  igc_fsdata() : FormatSpecificData(kFsIGC) {}
+
+  igc_fsdata* clone() const override
+  {
+    return new igc_fsdata(*this);
+  }
+
+  std::optional<T> enl; // Engine Noise Level
+  std::optional<T> tas; // True Airspeed
+  std::optional<T> vat; // Compensated variometer (total energy)
+  std::optional<T> oat; // Outside Air Temperature
+  std::optional<T> trt; // True Track
+  std::optional<T> gsp; // Ground Speed
+  std::optional<T> fxa; // Fix Accuracy
+  std::optional<T> siu; // Satellites In Use
+  std::optional<T> acz; // Z Acceleration
+  std::optional<T> gfo; // G Force?
+
+  bool set_value(IgcFormat::igc_ext_type_t type, int value) {
+    bool success = true;
+    switch (type) {
+      case IgcFormat::igc_ext_type_t::ext_rec_enl:
+        enl = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_tas:
+        tas = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_vat:
+        vat = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_oat:
+        oat = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_trt:
+        trt = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_gsp:
+        gsp = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_fxa:
+        fxa = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_siu:
+        siu = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_acz:
+        acz = value; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_gfo:
+        gfo = value; break;
+      default:
+        success = false;
+    }
+    return success;
+  }
+
+  // Not currently used, but already written and left for future use.
+  std::optional<int> get_value(IgcFormat::igc_ext_type_t defn_type) const {
+    std::optional<int> ret;
+    switch (defn_type) {
+      case IgcFormat::igc_ext_type_t::ext_rec_enl:
+        ret = enl; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_tas:
+        ret = tas; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_vat:
+        ret = vat; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_oat:
+        ret = oat; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_trt:
+        ret = trt; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_gsp:
+        ret = gsp; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_fxa:
+        ret = fxa; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_siu:
+        ret = siu; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_acz:
+        ret = acz; break;
+      case IgcFormat::igc_ext_type_t::ext_rec_gfo:
+        ret = gfo; break;
+      default:
+        fatal("Invalid igc_ext_type\n");
+        break;
+    }
+    return ret;
+  }
+  std::optional<int> get_value(KmlFormat::wp_field defn_type) const {
```

Comment:
In line with my comment https://github.com/GPSBabel/gpsbabel/pull/1054#issuecomment-1496723306 hopefully get_value can just return a double all the time.

I see alot of confusion about template parameters.  I would like to avoid creating templated functions, it's a bit more advanced and so far we are just breaking things and not adding any value.